File: api/web/models/run_log.py
"""APIs for RunLog"""
from typing import Any, Dict, Optional

from modules.types import PMObject


class RunLogInstance(PMObject):
    id: int
    created_at: str
    uuid: str

    run_from_deployment: bool

    input_register_name: Optional[str] = None
    inputs: Optional[Dict[str, Any]] = None
    raw_output: Optional[str] = None
    parsed_outputs: Optional[Dict[str, Any]] = None
    function_call: Optional[Dict[str, Any]] = None

    run_log_metadata: Optional[Dict[str, Any]] = None
    score: Optional[int] = None

    token_usage: Optional[Dict[str, Any]]
    latency: Optional[float] = None
    cost: Optional[float] = None

    version_uuid: str


class DeploymentRunLogViewInstance(PMObject):
    project_uuid: str

    prompt_model_uuid: str
    prompt_model_name: str

    prompt_model_version_uuid: str
    prompt_model_version: int

    created_at: str

    inputs: Optional[Dict[str, Any]] = None
    raw_output: Optional[str] = None
    parsed_outputs: Optional[Dict[str, Any]] = None
    function_call: Optional[Dict[str, Any]] = None

    token_usage: Optional[Dict[str, Any]] = None
    latency: Optional[float] = None
    cost: Optional[float] = None

    run_log_metadata: Optional[Dict[str, Any]] = None
    score: Optional[int] = None

    run_from_deployment: bool = True


class RunLogsCountInstance(PMObject):
    project_uuid: str
    count: int

File: src/types/RunLog.ts
// Interfaces for RunLog retrieval processes

/**
 * Interface for request to read PromptModelVersion's RunLogs.
 */
export interface ReadVersionRunLogsRequest {
  prompt_model_version_uuid: string;
}

/**
 * Interface for request to read Project's RunLogs.
 */
export interface ReadProjectRunLogsRequest {
  project_uuid: string;
  page: number;
  rows_per_page: number;
}

/**
 * Interface for request to read Project's RunLogs count.
 */
export interface ReadProjectRunLogsCountRequest {
  project_uuid: string;
}

/**
 * Interface for response of RunLogs count.
 */
export interface ReadProjectRunLogsCountResponse {
  project_uuid: string;
  count: number;
}

/**
 * General interface for RunLog.
 */
export interface RunLog {
  id: number;
  created_at: string;
  uuid: string;
  run_from_deployment: boolean;
  input_register_name: string;
  inputs: Record<string, any>;
  raw_output: string;
  parsed_outputs: Record<string, any>;
  function_call: Record<string, any>;
  run_log_metadata: Record<string, any>;
  score: number;
  token_usage: Record<string, any>;
  latency: number;
  cost: number;
  version_uuid: string;
}
